src/module/actor/base.ts

```diff
@@ -940,28 +965,49 @@ class ActorPF2e<TParent extends TokenDocumentPF2e | null = TokenDocumentPF2e | n
      * All attack rolls have the "all" and "attack-roll" domains and the "attack" trait,
      * but more can be added via the options.
      */
-    async getCheckRollContext<TStatistic extends StatisticCheck | StatisticModifier, TItem extends AttackItem | null>(
-        params: AttackRollContextParams<TStatistic, TItem>
-    ): Promise<AttackRollContext<this, TStatistic, TItem>> {
+    async getCheckContext<TStatistic extends StatisticCheck | StrikeData, TItem extends AttackItem | null>(
+        params: CheckContextParams<TStatistic, TItem>
+    ): Promise<CheckContext<this, TStatistic, TItem>> {
         const context = await this.getRollContext(params);
         const targetActor = context.target?.actor;
         const rangeIncrement = context.target?.rangeIncrement ?? null;
 
         const rangePenalty = calculateRangePenalty(this, rangeIncrement, params.domains, context.options);
         if (rangePenalty) context.self.modifiers.push(rangePenalty);
 
-        return {
-            ...context,
-            dc: targetActor?.attributes.ac
-                ? {
-                      scope: "attack",
-                      slug: "ac",
-                      statistic:
-                          targetActor.attributes.ac instanceof StatisticModifier ? targetActor.attributes.ac : null,
-                      value: targetActor.attributes.ac.value,
-                  }
-                : null,
-        };
+        const dcData = ((): CheckDC | null => {
+            const { domains, targetedDC } = params;
+            const scope = domains.includes("attack") ? "attack" : "check";
+
+            switch (targetedDC) {
+                case "ac":
+                case "armor":
+                    return targetActor?.attributes.ac
+                        ? {
+                              scope,
+                              slug: "ac",
+                              statistic:
+                                  targetActor.attributes.ac instanceof StatisticModifier
+                                      ? targetActor.attributes.ac
+                                      : null,
+                              value: targetActor.attributes.ac.value,
+                          }
+                        : null;
+                case "fortitude":
+                case "reflex":
+                case "will": {
+                    const statistic = targetActor?.saves?.[targetedDC]?.dc;
+                    return statistic ? { scope, statistic, slug: targetedDC, value: statistic.value } : null;
+                }
+                // A skill DC
+                default: {
+                    const statistic = targetActor?.skills?.[targetedDC]?.dc;
+                    return statistic ? { scope, statistic, slug: targetedDC, value: statistic.value } : null;
+                }
+            }
+        })();
```

Comment:
How well would it work to use getStatistic() for this? I can imagine ac/armor won't work well, but the rest seem doable by that method at a glance.